tmt/utils.py

```diff
@@ -2010,6 +2010,29 @@ def public_git_url(url: str) -> str:
     return url
 
 
+def public_web_url(fmf_id: 'tmt.base.FmfId', path: str) -> str:
+    """
+    Convert a public git url into a clickable web url format
+
+    Compose a clickable link from git url, ref and path to file
+    for most common git servers
+    """
+    assert fmf_id.url is not None
+    assert fmf_id.ref is not None  # narrow type
+    if any(server in fmf_id.url for server in ['github', 'gitlab']):
+        web_url: str = fmf_id.url.replace('.git', f'/tree/{fmf_id.ref}')
+
+    if 'pkgs.devel' in fmf_id.url:
+        web_url = fmf_id.url.replace('git', 'https').replace('.com', '.com/cgit')
```

Comment:
Looks ok, but a little bit unsafe? Perhaps, would it make sense to use regexp to ensure that for example `git` is only replaced at the beginning of the string?

```
>>> tmt.utils.public_web_url(tmt.base.FmfId(url="git://pkgs.devel.redhat.com/tests/git", ref="master"), path="/Sanity/upstream")
'https://pkgs.devel.redhat.com/cgit/tests/https/tree/Sanity/upstream'
```

Seems this would not work for the `git` component.

Reply:
Don't think pulling out the big boy re.sub is necessary here. Matching 'git://' now instead.